tests/components/withings/test_sensor.py

```diff
@@ -0,0 +1,748 @@
+                    1310,
+                    1410
+                ),
+            )
+        ]
+    })
+
+    await assert_health_sensor_equals(
+        2.7,
+        const.MEAS_SLEEP_WAKEUP_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        4.3,
+        const.MEAS_SLEEP_LIGHT_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        6.0,
+        const.MEAS_SLEEP_DEEP_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        7.7,
+        const.MEAS_SLEEP_REM_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        160.0,
+        const.MEAS_SLEEP_WAKEUP_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        260.0,
+        const.MEAS_SLEEP_LIGHT_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        360,
+        const.MEAS_SLEEP_DEEP_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        460.0,
+        const.MEAS_SLEEP_REM_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        560.0,
+        const.MEAS_SLEEP_WAKEUP_COUNT,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        11.0,
+        const.MEAS_SLEEP_TOSLEEP_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        12.7,
+        const.MEAS_SLEEP_TOWAKEUP_DURATION_HOURS,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        660.0,
+        const.MEAS_SLEEP_TOSLEEP_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        760.0,
+        const.MEAS_SLEEP_TOWAKEUP_DURATION_MINUTES,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        860.0,
+        const.MEAS_SLEEP_HEART_RATE_AVERAGE,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        960.0,
+        const.MEAS_SLEEP_HEART_RATE_MIN,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        1060.0,
+        const.MEAS_SLEEP_HEART_RATE_MAX,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        1160.0,
+        const.MEAS_SLEEP_RESPIRATORY_RATE_AVERAGE,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        1260.0,
+        const.MEAS_SLEEP_RESPIRATORY_RATE_MIN,
+        data_manager
+    )
+    await assert_health_sensor_equals(
+        1360.0,
+        const.MEAS_SLEEP_RESPIRATORY_RATE_MAX,
+        data_manager
+    )
+
+
+async def assert_health_sensor_equals(
+        expected,
+        measure: str,
+        data_manager: WithingsDataManager
+):
+    """Assert the state of a withings sensor."""
+    sensor = WithingsHealthSensor(
+        data_manager,
+        WITHINGS_MEASUREMENTS_MAP[measure]
+    )
+
+    await sensor.async_update()
+    assert sensor.state == expected, \
```

Comment:
This test helper should be rewritten as we've discussed. It should set up the component which sets up the platform, adds the entity, gets the state from the core state machine and then asserts the state and state attributes of the entity.

Reply:
Sounds like higher level integration tests are what's desired here. I've updated the tests accordingly.